### PR TITLE
Work on making installer cross platform

### PR DESCRIPTION
**Pull request by mirkojotic**
- Added windows as a platform detection option
- Handle extraction of .zip file

Solves #615

### CODE REVIEW

**Comment by mirkojotic** [comment]
Hey @ry first I've got it to build on windows :tada: and it's building on linux (Travis and locally) always.

But it is not building for OSX - due to:
```
{
  "message": "API rate limit exceeded for 207.254.16.35. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)",
  "documentation_url": "https://developer.github.com/v3/#rate-limiting"
}
```
The reason why OSX always fails is that it's slower and runs after linux. I'm open for suggestions :D

**Comment by ry** [comment]
@mirkojotic That's annoying. How about we work around it by not hitting the API server, and instead using a regex to extract the tag from https://github.com/repos/denoland/deno/releases/latest

(dear github people - it would be nice if we hit https://github.com/denoland/deno/releases/download/latest/deno_osx_x64.gz  instead of having to look up the tag.)

**Comment by mirkojotic** [comment]
@ry
Lol. Yes that was the first thing I tried 😁 it would be really nice.

Yes. I agree with regex approach! I'll continue working on that. 

Thanks!

File: tools/install.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+home = os.path.expanduser("~")
+
+def get_latest_url():
+
+    res = urlopen(releases_url_html)
+
+    html = res.read().decode('utf-8')
+
+    urls = re.findall(r'href=[\'"]?([^\'" >]+)', html)
+
```
Comment (ry) [inline tools/install.py]: There's a lot of blank lines - seems unnecessary for reading - can you condense it a bit?

File: tools/install.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    matching = [u for u in urls if filename in u]
+
+    if len(matching) != 1:
+        print("Bad download url")
+        print("urls", urls)
+        print("matching", matching)
+        sys.exit(1)
+
+    return GITHUB_URL + matching[0]
```
Comment (ry) [inline tools/install.py]: I think it would be more readable if you 
```
return "https://github.com" + matching[0]
```

File: tools/install.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    latest_url = get_latest_url()
+    latest_fn = dlfile(latest_url)
+
+    bin_dir = deno_bin_dir()
+    exe_fn = os.path.join(bin_dir, "deno")
+
+    if "zip" in latest_fn:
+        with ZipFile(latest_fn, 'r') as z:
+            with open(exe_fn, 'wb+') as exe:
+                exe.write(z.read('deno.exe'))
```
Comment (ry) [inline tools/install.py]: Nice!

File: tools/install.py (hunk 2 above)
Comment (mirkojotic) [reply tools/install.py]: Done!

File: tools/install.py (hunk 1 above)
Comment (mirkojotic) [reply tools/install.py]: Done! I've still have a couple of blank lines but less :D

**Pull request merged**